app/controllers/users_controller.rb

```diff
@@ -21,7 +26,7 @@ def show
   private
 
   def user_params
-    params.permit(:name, :address, :city, :state, :zip, :email, :password)
+    params.require(:user).permit(:name, :address, :city, :state, :zip, :email, :password)
```

Comment:
I think I understand why this is happening but just want to verify. Is this because we are using the form for instead of the form tag? I think the form for changes the params so that the input from the form is a hash value of params[:user].

Reply:
Im actually glad you pointed this out as I'd forgotten about it. Its the product of several stackoverflows that I pieced together to get the fields to retain their info. It was late at the time but Im pretty sure it didn't work if I removed that bit. Unfortunately I cannot specifically say what it is doing.

Reply:
It might be unnecessary I'll test it out again

Reply:
Just tested it out and sure enough the fields do not retain the user input when the page reloads if `.require(:user)` is removed from line 29

Reply:
Just so you know, this was part of my previous pr for story 12